src/main/java/com/celements/cleverreach/CleverReachRest.java

```diff
@@ -222,10 +286,48 @@ Response sendRequest(String path, Object data, String authHeader, SubmitMethod m
     return sendRequest(path, data, authHeader, method, connection.getBaseUrl());
   }
 
+  boolean isReadyToSendPut(Response response) {
+    if ((response != null) && response.hasEntity()) {
+      String content = response.readEntity(String.class);
+      try {
+        Optional<ResponseBodyObj> resultObj = Arrays.asList(new ObjectMapper()
+            .readValue(content, ResponseBodyObj.class)).stream().findFirst();
+        return resultObj.isPresent() && "1".equals(resultObj.get().value);
+      } catch (IOException ioe) {
+        LOGGER.warn("Parsing CleverReach response to JSON failed. Content [{}]", content, ioe);
+      }
+    } else {
+      LOGGER.warn("Mailing update failed with response [{}] and response hasEntity [{}]", response,
+          (response != null) && response.hasEntity());
+    }
+    return false;
+  }
+
+  boolean isReadyToSendGet(Response response, ServerClass server) {
+    String readyToSend = "ready_to_send_" + server.toString().toLowerCase();
+    if ((response != null) && response.hasEntity()) {
+      String content = response.readEntity(String.class);
+      try {
+        Optional<ResponseBodyObj> resultObj = Arrays.asList(new ObjectMapper()
+            .readValue(content, ResponseBodyObj[].class))
+            .stream()
+            .filter(ele -> readyToSend.equals(ele.name))
+            .findFirst();
+        return resultObj.isPresent() && "1".equals(resultObj.get().value);
+      } catch (IOException ioe) {
+        LOGGER.warn("Parsing CleverReach response to JSON failed. Content [{}]", content, ioe);
+      }
+    } else {
+      LOGGER.warn("Mailing update failed with response [{}] and response hasEntity [{}]", response,
+          (response != null) && response.hasEntity());
```

Comment:
```suggestion
      LOGGER.warn("Mailing update failed with response [{}]", response);
```
`hasEntity` is always false here